app/api/chat/route.ts

```
@@ -0,0 +1,479 @@
+    let completionTokens = 0
+
+    switch (modelConfig.provider) {
+      case 'OPENAI':
+        response = await callOpenAICompatible(apiKey, message, model, conversationHistory, modelConfig.endpoint)
+        promptTokens = response.usage?.prompt_tokens || 0
+        completionTokens = response.usage?.completion_tokens || 0
+        break
+      
+      case 'XAI':
+        // X.AI uses 'grok-2-1212' but their API expects 'grok-2'
+        const xaiModel = model === 'grok-2-1212' ? 'grok-2' : model
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)